aten/src/ATen/native/mps/operations/View.mm

```diff
@@ -49,7 +54,7 @@
                                                                                shape: inputShape
                                                                             dataType: inputType] autorelease];
     if (needsScatter) {
-      feeds[cachedGraph->updatesTensor] = [[[MPSGraphTensorData alloc] initWithMTLBuffer: sourceBuffer
+      feeds[cachedGraph->updatesTensor] = [[[MPSGraphTensorData alloc] initWithMTLBuffer: (updatesBuffer != nil) ? updatesBuffer : sourceBuffer
                                                                                    shape: getMPSShape(src.numel())
```

Comment:
Is the shape of this updatesBuffer always the same as `getMPSShape(src.numel())`? I mean, considering the pointer arithmetic you performed on it in `host_src + (src_.storage_offset() * src_.itemsize())` before passing it to runViewGraph..

Reply:
Yes, they will always match. `src.numel()` also takes into account `storage_offset()` when returning  the total number of elements a tensor has. The pointer arithmetic is needed for view tensors where the data of the tensor doesn't always start from offset 0